node/cli/src/command.rs

```diff
@@ -1,168 +1,189 @@
+                    } = new_partial(&config)?;
+                    Ok((cmd.run(client, config.database), task_manager))
+                })
+            }
+            Some(Subcommand::ExportState(cmd)) => {
+                let runner = cli.create_runner(cmd)?;
+                runner.async_run(|config| {
+                    let PartialComponents {
+                        client,
+                        task_manager,
+                        ..
+                    } = new_partial(&config)?;
+                    Ok((cmd.run(client, config.chain_spec), task_manager))
+                })
+            }
+            Some(Subcommand::ImportBlocks(cmd)) => {
+                let runner = cli.create_runner(cmd)?;
+                runner.async_run(|config| {
+                    let PartialComponents {
+                        client,
+                        task_manager,
+                        import_queue,
+                        ..
+                    } = new_partial(&config)?;
+                    Ok((cmd.run(client, import_queue), task_manager))
+                })
+            }
+            Some(Subcommand::PurgeChain(cmd)) => {
+                let runner = cli.create_runner(cmd)?;
+                runner.sync_run(|config| cmd.run(config.database))
+            }
+            Some(Subcommand::Revert(cmd)) => {
+                let runner = cli.create_runner(cmd)?;
+                runner.async_run(|config| {
+                    let PartialComponents {
+                        client,
+                        task_manager,
+                        backend,
+                        ..
+                    } = new_partial(&config)?;
+                    Ok((cmd.run(client, backend), task_manager))
+                })
+            }
+            Some(Subcommand::Benchmark(cmd)) => {
+                if cfg!(feature = "runtime-benchmarks") {
+                    let runner = cli.create_runner(cmd)?;
+
+                    runner.sync_run(|config| cmd.run::<Block, ExecutorDispatch>(config))
+                } else {
+                    Err(
+                        "Benchmarking wasn't enabled when building the node. You can enable it with \
+				     `--features runtime-benchmarks`."
+                            .into(),
+                    )
+                }
+            }
+            None => {
+                let runner = cli.create_runner(&cli.run)?;
+                runner.run_node_until_exit(|config| async move {
+                    new_full(config).map_err(sc_cli::Error::Service)
+                })
+            }
+        }
     }
 }
 
-#[allow(clippy::borrowed_box)]
-fn extract_genesis_wasm(chain_spec: &Box<dyn sc_service::ChainSpec>) -> Result<Vec<u8>> {
-    let mut storage = chain_spec.build_storage()?;
-
-    storage
-        .top
-        .remove(sp_core::storage::well_known_keys::CODE)
-        .ok_or_else(|| "Could not find wasm file in genesis state!".into())
-}
-
 #[cfg(feature = "with-parachain-runtime")]
```

Comment:
I think this is definitely a good start, i didnt know we already did some of this 😄 

Although i might say this is largely one and the same still, e.g:

https://github.com/t3rn/t3rn/blob/bca51132217382092594f8beca6bf52f26588e7e/node/cli/src/command.rs#L217

and https://github.com/t3rn/t3rn/blob/bca51132217382092594f8beca6bf52f26588e7e/node/cli/src/command.rs#L25

They are the same minus the string, so im sure you can get away with something like this:
```rust
fn impl_name() -> String {
            #[cfg(feature = "with-parachain-runtime")]
            "Circuit Parachain Collator".into()
            #[cfg(feature = "with-standalone-runtime")]
            "Circuit Node".into()
}
```
Providing that we always use at least one of these features, although to be devils avocate, it does increase complexity with build tools etc.

This might help us in the future in terms of cognitive load (i think substrate loves giant files with no real module XD)

This is not for this PR, i think its just a good optimisation we can be aware of in the future as we go along, as it would be annoying to do it *all* right now

Reply:
I see your point, this is how it used to be, but in this case, the SubstrateCli trait implementation would have to be the only one extracted outside the 2 modules, and then these feature flags would be needed for other functions and contstants like `load_spec`, `VERSION` etc. I feel like both versions are a bit messy and decided to keep it like this for now just to have everything grouped together.